crates/rolldown/src/bundle/bundle_factory.rs

```diff
@@ -26,16 +28,23 @@ pub struct BundleFactoryOptions {
 }
 
 pub struct BundleFactory {
+  pub plugin_driver_factory: PluginDriverFactory,
   pub fs: OsFileSystem,
   pub options: SharedOptions,
   pub resolver: SharedResolver,
   pub file_emitter: SharedFileEmitter,
-  pub plugin_driver: SharedPluginDriver,
   /// Warnings collected during bundle factory creation.
   /// These warnings are transferred to the first created `Bundle` via `create_bundle()` or `create_incremental_bundle()`.
   pub warnings: Vec<BuildDiagnostic>,
   pub session: rolldown_debug::Session,
   pub(crate) _log_guard: Option<Box<dyn Any + Send>>,
+  pub last_bundle_context: Option<BundleContext>,
+
+  // Used to share module info across multiple plugin drivers for incremental builds
+  module_infos_for_incremental_build: SharedModuleInfoDashMap,
```

Comment:
If we don't manage it,

https://github.com/rolldown/rolldown/blob/068cfaace69691040834708b10d786b79788fbda/packages/test-dev-server/tests/fixtures/defer-sync-scan-data/dev.config.mjs#L19-L26

will throw an error.

In the second `transform` call for `main.js`, `this.getModuleInfo` gets nothing, becuase we create totoal new context for the hmr build.

This applies to watch mode too, but we don't have watch mode test for testing this. Glad we catch it here.